### DISCUSSION TITLE
What's the default behaviour when all connections are currently in use?

### DISCUSSION THREAD

**Discussion by Haiz14** [sqlalchemy/sqlalchemy#11467] [Usage Questions] [answered]
Suppose my postgres db has 30 connectioms max. And all of them are in use. Then an update query was called by an api endpoint. What will happen? 
- Will sqlalchemy directly raise error ? 
- wait till a certain time for connections to be released and then raise error if no connections were found

**Comment by CaselIT** [comment, marked as answer]
Hi,

Depends on what you mean by 30 connections.
If those are the one configured on the postgresql server, then when the driver will try to connect it will wait for a timeout then raise an exception. Loot at the documentation of the driver in use on how to customise the timeout 

If those 30 connection are the configuration of the queue pool then a similar behaviour happens. The default timeout in this case is 30 seconds https://docs.sqlalchemy.org/en/20/core/pooling.html#sqlalchemy.pool.QueuePool.params.timeout

**Comment by Haiz14** [comment]
Thanks
